src/app/modules/test-case-service/default.test.case.service.ts

```diff
@@ -0,0 +1,81 @@
+import { TestCaseServiceConfig } from './test.case.service.config';
+import { Injectable, Injector } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { MessagingService } from '@testeditor/messaging-service';
+
+// code duplication with persistence service and test-editor-web, removal planned with next refactoring
+const HTTP_CLIENT_NEEDED = 'httpClient.needed';
+const HTTP_CLIENT_SUPPLIED = 'httpClient.supplied';
+
+// API provided by service backend
+export class CallTreeNode {
+  displayName: string;
+  children: CallTreeNode[];
+}
+
+export abstract class TestCaseService {
+  abstract getCallTree(path: string, onResponse?: (some: any) => void, onError?: (error: any) => void): void;
```

Comment:
shouldn't the signature of onResponse be `(status: CallTreeNode) => void`? Callers should be aware of the type to be able to process the tree data.